### ISSUE TITLE
[2.0.2] Upload problems: File changes progress(?)

### ISSUE THREAD

**Issue by Dscheydschey** [owncloud/client#4240]
I am a user of owncloud including the actuell webserver version 8.2.1 and desktop client 2.0.2.
I didn't have any problems with uploads using desktop client-versions up to 1.8.4 .5226.
But problems appeared from update to current version 2.0.2.5569. This version shows problems I do not understand: 
Upload starts and goes up to about 13 MB (filesize is 16 MB), falls back to about 8.x MB, goes up again and the upload get interrupted just before it finished. After a few seconds update starts again and got finally finished. This happens each time when a new version of the file should be uploaded.

As I said before:  Uploads worked correctly with the 1.8.4 client and OC 8.2.1! 
What is wrong here? Maybe there is somebody, who can help solving this problem? 

Thank you very much

**Comment by guruz** [comment]
Which platform?
If Linux: Which Qt version? (see it in the client settings dialog)

> and the upload get interrupted just before it finished. 

What does that mean? What error do you see?
We cannot deduce from this description what is going on :)

**Comment by Dscheydschey** [comment]
Hi,
I work under Windows 10, where the client is installed. OC is on my webspace-server at the provider 1und1.de installed. I use oc just to sync one single file (filesize 16 MB). Upload starts while I am working and changing the file. But this was not a problem in the former versions.
All uploads and downloads worked correctly under clientversion 1.8.4 and oc-version 8.2.1

What happens: you can see in the progress bar that the upload
[Owncloudlog.docx](https://github.com/owncloud/client/files/49924/Owncloudlog.docx)
[activity window.docx](https://github.com/owncloud/client/files/49926/activity.window.docx)

 continues up to 11-13 MB (of 16MB); then the progress bar falls back to about 8 MB, goes up again and then sometimes the upload breaks down completely, the other time it finishes finally.
That makes me nerveous!!!
Failurereport says: "couldn't complete upload becaus of a failure"  
See also attached files 

Hope you can solve my problem. Thank you

**Comment by Dscheydschey** [comment]
sorry, I didn' want to close the issue

**Comment by dragotin** [comment]
Please write a log file with the client (see client doc how to do that) and put it to gist.

**Comment by danimo** [comment]
Can you please not use docx for screenshots? you can copy and paste them right into the comment field. Also, please provide the log file: https://raw.githubusercontent.com/owncloud/client/master/issue_template.md

**Comment by Dscheydschey** [comment]
Hi,

 

Sorry I am not very familiar with all these things. I am just a user of owncloud but I don't have any knowledge about things behind.

Where can I find  client doc? What I sent to you as "activity window.docx" was in my eyes the client  log??!! I copied it from that client window.

 

What do I have to do with the Link  <https://raw.githubusercontent.com/owncloud/client/master/issue_template.md> https://raw.githubusercontent.com/owncloud/client/master/issue_template.md Daniel sent to me? Should I answer all these questions?

I don't know how to get all these information.

 

Kind regards 

 

Jakob Jambach

 

Von: Klaas Freitag [mailto:notifications@github.com] 
Gesendet: Mittwoch, 2. Dezember 2015 18:00
An: owncloud/client
Cc: Dscheydschey
Betreff: Re: [client] [2.0.2] Upload problems: File changes progress(?) (#4240)

 

Please write a log file with the client (see client doc how to do that) and put it to gist.

**Comment by Dscheydschey** [comment]
Hi,

Here comes the logfile content, I hope you can identify the problem:

12-03 09:37:02:068 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:02:069 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:02:069 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_disconnected() 0 QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0
12-03 09:37:13:364 0x4c17738 OCC::Folder::slotRunEtagJob: * Trying to check "Datenbank" for changes via ETag check. (time since last sync: 704 s)
12-03 09:37:13:364 0x4c17738 OCC::FolderMan::slotRunOneEtagJob: Scheduling "Datenbank" to check remote ETag
12-03 09:37:13:364 0x4c17738 OCC::AbstractNetworkJob::start: !!! OCC::RequestEtagJob created for "https://www.kleingartenverein-ufg.de/owncloud" + "/"
12-03 09:37:13:676 0x4c17738 OCC::Folder::etagRetreived: * Compare etag with previous etag: last: ""565f208d58e3d""7d082d0061d624581f273d2cc86a6bbf"" , received: ""565f208d58e3d""7d082d0061d624581f273d2cc86a6bbf""
12-03 09:37:13:692 0x4c17738 OCC::FolderMan::slotRunOneEtagJob: No more remote ETag check jobs to schedule.
12-03 09:37:15:688 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:15:688 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:15:688 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_disconnected() 0 QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0
12-03 09:37:31:526 0x4c17738 unknown: Failed to resolve EGL function eglGetPlatformDisplayEXT (Die angegebene Prozedur wurde nicht gefunden.)
12-03 09:37:31:526 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): Could not initialize EGL display: error 0x3001

12-03 09:37:31:526 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): When using ANGLE, check if d3dcompiler_4x.dll is available
12-03 09:37:31:593 0x4c17738 unknown: Failed to resolve EGL function eglGetPlatformDisplayEXT (Die angegebene Prozedur wurde nicht gefunden.)
12-03 09:37:31:593 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): Could not initialize EGL display: error 0x3001

12-03 09:37:31:593 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): When using ANGLE, check if d3dcompiler_4x.dll is available
12-03 09:37:31:733 0x4c17738 OCC::ConnectionValidator::checkAuthentication: # Check whether authenticated propfind works.
12-03 09:37:31:733 0x4c17738 OCC::AbstractNetworkJob::start: !!! OCC::PropfindJob created for "https://www.kleingartenverein-ufg.de/owncloud" + "/"
12-03 09:37:34:124 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:34:124 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:34:124 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_disconnected() 0 QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0
12-03 09:37:43:366 0x4c17738 OCC::Folder::slotRunEtagJob: * Trying to check "Datenbank" for changes via ETag check. (time since last sync: 734 s)
12-03 09:37:43:366 0x4c17738 OCC::FolderMan::slotRunOneEtagJob: Scheduling "Datenbank" to check remote ETag
12-03 09:37:43:366 0x4c17738 OCC::AbstractNetworkJob::start: !!! OCC::RequestEtagJob created for "https://www.kleingartenverein-ufg.de/owncloud" + "/"
12-03 09:37:43:726 0x4c17738 OCC::Folder::etagRetreived: * Compare etag with previous etag: last: ""565f208d58e3d""7d082d0061d624581f273d2cc86a6bbf"" , received: ""565f208d58e3d""7d082d0061d624581f273d2cc86a6bbf""
12-03 09:37:43:741 0x4c17738 OCC::FolderMan::slotRunOneEtagJob: No more remote ETag check jobs to schedule.
12-03 09:37:45:726 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:45:726 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_error(QAbstractSocket::SocketError) QAbstractSocket::RemoteHostClosedError QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0 QAbstractSocket::ConnectedState
12-03 09:37:45:726 0x7024d40 unknown: void QHttpNetworkConnectionChannel::_q_disconnected() 0 QSslSocket(0x7024dd0) QObject(0x0)  0 false 3 0
12-03 09:37:53:374 0x4c17738 unknown: Failed to resolve EGL function eglGetPlatformDisplayEXT (Die angegebene Prozedur wurde nicht gefunden.)
12-03 09:37:53:374 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): Could not initialize EGL display: error 0x3001

12-03 09:37:53:374 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): When using ANGLE, check if d3dcompiler_4x.dll is available
12-03 09:37:53:437 0x4c17738 unknown: Failed to resolve EGL function eglGetPlatformDisplayEXT (Die angegebene Prozedur wurde nicht gefunden.)
12-03 09:37:53:437 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): Could not initialize EGL display: error 0x3001

12-03 09:37:53:437 0x4c17738 unknown: static QWindowsEGLStaticContext* QWindowsEGLStaticContext::create(): When using ANGLE, check if d3dcompiler_4x.dll is available

**Comment by Dscheydschey** [comment]
In addition to the comment above (clientlog) here is what owncloud.log (server) now shows:

{"reqId":"Vl8apAouAwwAADHbe2YAAABg","remoteAddr":"217.249.166.69","app":"PHP","message":"Array to string conversion at \/homepages\/44\/d84325047\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-12-02T16:21:56+00:00"}
{"reqId":"Vl8fhgouAwwAAHe@BwUAAAAh","remoteAddr":"217.249.166.69","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 5242880 got 3604480\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(101): OC\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #286)\\n#1 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(115): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #286)\\n#2 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('pc_klein_DB_4_0...', Resource id #286)\\n#3 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('pc_klein_DB_4_0...', Resource id #286, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/remote.php(137): require_once('\\\/homepages\\\/44\\\/d...')\\n#10 {main}\",\"File\":\"\\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php\",\"Line\":361}","level":4,"time":"2015-12-02T16:44:37+00:00"}
{"reqId":"Vl8fhwouAwwAAHIIEj8AAABI","remoteAddr":"217.249.166.69","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 5242880 got 4456448\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(101): OC\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #286)\\n#1 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(115): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #286)\\n#2 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('pc_klein_DB_4_0...', Resource id #286)\\n#3 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('pc_klein_DB_4_0...', Resource id #286, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/remote.php(137): require_once('\\\/homepages\\\/44\\\/d...')\\n#10 {main}\",\"File\":\"\\\/homepages\\\/44\\\/d84325047\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php\",\"Line\":361}","level":4,"time":"2015-12-02T16:44:37+00:00"}
{"reqId":"VmAkRQouAwwAAGoiXokAAAAo","remoteAddr":"91.4.97.173","app":"PHP","message":"Array to string conversion at \/homepages\/44\/d84325047\/htdocs\/owncloud\/lib\/private\/template\/functions.php#36","level":3,"time":"2015-12-03T11:15:18+00:00"}

**Comment by Dscheydschey** [comment]
Hi again,
I tried different scenarios while the sync process and I think I found out at least one of the reasons of the described problem.
The described interruptions of the sync progress happen, when the file, which is to sync is changed by another program while the sync process.

When I open the sync file (MS Access-file) with this other program the time stamp changes and client starts immediately with the sync process (no other changes in the file!!). 
In my opinion former client versions started sync process when the syncfile was closed again by the other programm.  But I'm not sure.

How can I make the client to start the sync process after finishing my work? Or filechanges during the sync process mustn't interfere this process?
Is there any solution for this?

Thanks for your help
